src/components/layout/inputs/DurationInput/DurationInput.jsx

```diff
@@ -0,0 +1,113 @@
+import PropTypes from 'prop-types'
+import React, { useCallback, useState } from 'react'
+
+import InputError from '../Errors/InputError'
+
+const translateMinutesToHours = durationInMinutes => {
+  if (durationInMinutes === null) return ''
+  const hours = Math.floor(durationInMinutes / 60)
+  const minutes = durationInMinutes % 60
+  return `${hours}:${minutes}`
+}
+
+const validHoursDuration = durationInHours => durationInHours.match(/[0-9]*:[0-5][0-9]/)
+
+const hasFinishedWritingHours = durationInHours => durationInHours.match(/[0-9]+:/)
+
+const DurationInput = ({
+  disabled,
+  error,
+  label,
+  name,
+  onChange,
+  required,
+  subLabel,
+  initialDurationInMinutes,
+}) => {
+  const [durationInHours, setDurationInHours] = useState(
+    translateMinutesToHours(initialDurationInMinutes)
+  )
+
+  const onDurationChange = useCallback(
+    event => {
+      const updatedHoursDuration = event.target.value
+      let correctedHoursDuration
+
+      if (hasFinishedWritingHours(updatedHoursDuration)) {
+        if (updatedHoursDuration.match(/[0-9]+:[0-5][0-9]/)) {
+          correctedHoursDuration = updatedHoursDuration.match(/[0-9]+:[0-5][0-9]/)[0]
+          setDurationInHours(correctedHoursDuration)
+        } else if (updatedHoursDuration.match(/[0-9]+:[0-5]/)) {
+          correctedHoursDuration = updatedHoursDuration.match(/[0-9]+:[0-5]/)[0]
+          setDurationInHours(correctedHoursDuration)
+        } else {
+          correctedHoursDuration = updatedHoursDuration.match(/[0-9]+:/)[0]
+          setDurationInHours(correctedHoursDuration)
+        }
+      } else {
+        correctedHoursDuration = updatedHoursDuration.match(/[0-9]*/)[0]
+        setDurationInHours(correctedHoursDuration)
+      }
+
+      if (validHoursDuration(correctedHoursDuration)) {
+        const [updatedHours, updatedMinutes] = correctedHoursDuration.split(':')
+        const updatedDurationInMinutes = parseInt(updatedHours) * 60 + parseInt(updatedMinutes)
+        onChange(updatedDurationInMinutes)
+      } else if (correctedHoursDuration === '') {
+        onChange(null)
+      }
+    },
+    [onChange]
+  )
+
+  return (
+    <label className="input-time">
+      <div className="labels">
+        {label}
+        <span className="itime-sub-label">
+          {subLabel}
+        </span>
+      </div>
+      <span className={`itime-input-container ${error ? 'error' : ''}`}>
```

Comment:
```suggestion
      <span className={`itime-field-container ${error ? 'error' : ''}`}>
```
Juste parce qu'au départ ça m'a perturbé de voir `itime`, après je me suis dit que c'était pour `input-time` puis j'ai vu `itime-input` et là j'ai plus compris ;)